docs/index.md

```diff
@@ -51,20 +51,21 @@ Usage:
   dashboard-linter lint [dashboard.json] [flags]
 
 Flags:
-  -c, --config string   path to a configuration file
-      --fix             automatically fix problems if possible
-  -h, --help            help for lint
-      --strict          fail upon linting error or warning
-      --verbose         show more information about linting
+  -c, --config string         path to a configuration file
+      --fix                   automatically fix problems if possible
+  -h, --help                  help for lint
+  -m, --matcher stringArray   matcher required to be present in all selectors, e.g. 'instance=~"$instance"' or 'cluster="$cluster"', can be specified multiple times (default ["instance=~""$instance""","job=~""$job"""])
+      --stdin                 read from stdin
+      --strict                fail upon linting error or warning
+      --verbose               show more information about linting
 ```
 
 # Rules
 
 The linter implements the following rules:
 
 * [template-datasource-rule](./rules/template-datasource-rule.md) - Checks that the dashboard has a templated datasource.
-* [template-job-rule](./rules/template-job-rule.md) - Checks that the dashboard has a templated job.
-* [template-instance-rule](./rules/template-instance-rule.md) - Checks that the dashboard has a templated instance.
+* [template-variable-matchers-rule](./rules/template-variable-matchers-rule.md) - Checks that the dashboard has a template variable for required matchers that use variables.
```

Comment:
Also in the interest of providing a smooth transition for existing users, let's make the addition of this rule be conditional.

If the user has specified the matchers in the config file (see above) then the rule will be run. Else it will not.

I need to add a deprecation mechanism for rules, and probably a flag for "new" or "experimental" rules. As well as properly versioning/releasing the dashboard linter (see #150). This would allow users to plan their migrations to a new version of the dashboard linter.

I can work on some issue(s) to describe those mechanisms, and build them separately. For now, just having this new rule run conditionally, only when configuration for it exists, will allow us to get it merged.